src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -95,6 +95,18 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
           )
         )
     }
+
+    // Fallback on the userId if we can't find the user in sam, this vastly simplifies the logic in the rest of the code
+    samUser.recoverWith({
+      case _: Exception =>
+        logger.warn(s"Unable to find user in sam, falling back on userId: $userId")
+        val dummySamUser = new sam.model.User()
+        dummySamUser.setGoogleSubjectId(userId)
+        dummySamUser.setAzureB2CId(userId)
+        dummySamUser.setId(userId)
+
+        Future.successful(dummySamUser)
+    })
```

Comment:
This is smelly but this whole sam api call is smelly. If I dont fallback in this way I have to handle this in like 3-4 separate places.

Reply:
Also I tried `case e: ApiException if e.getStatusCode == 404` but that didnt seem to be working so i still need to give that a shot again